zebra/zebra_vxlan.c

```diff
@@ -3839,16 +3840,51 @@ static struct interface *zvni_map_to_svi(vlanid_t vid, struct interface *br_if)
 	return tmp_if;
 }
 
+static int zvni_map_to_macvlan_ns(struct ns *ns,
```

Comment:
I did not do it because natively, the code was already duplicated.
so here is the list:
(1)- zvni from svi => look for vxlan interface with same bridge that is not vlan awre
(2)- zvni map vlan =>  look for vxlan interface with same bridge and vlan id 
(3)- zvni map svi => look for vlan interface with same vlan id
(4)- zvni map mac vlan => look for macvlan interface with same link interface

it is possible to do a merge of some functions:
- (1) and (2)
- eventually apply to all ( include (3) and (4))

I tend to think it is better to not do this as it may bring complexity to maintain this single function.
however, thanks to confirm me you think it is better to do this, please.

Reply:
Sri, any feedback ?

Reply:
I am specifically referring to zvni_map_to_svi, zvni_map_to_svi_ns, zvni_map_to_macvlan and zvni_map_to_macvlan_ns functions. My earlier comment may not be very clear with respect to this.